ddtrace/tracer/tracer_test.go

```diff
@@ -147,7 +147,6 @@ func TestTracerStart(t *testing.T) {
 	t.Run("deadlock/direct", func(t *testing.T) {
 		tr, _, stop := startTestTracer()
 		defer stop()
-		go tr.worker()
```

Comment:
Not sure why this was here to begin with.
This starts a second tracer worker, which is invalid.

The reason this wasn't causing failures before was that the previous version of `Stop()` would only shut down one of the workers, and subsequent calls to `Stop()` wouldn't do anything, so the second worker never had the opportunity to try shutting down and panic.

Reply:
The usual answer for "why is this here" in a test is that it was needed in the past, to lock in the behavior of the earlier implementation
If it's not needed now, then it's a target for fixes/changes/removal.

I don't know the actual history of this deadlock test, but if the implementation changed enough to need fixing the tests, that's fine.
Just be sure the test is still doing what it was intended for (or guess as best as possible), and that the "fixes" accurately lock in the new behavior that is not covered by other tests.